### PR TITLE
sys-process/systemd-cron: Fix systemd generator path

### PR DESCRIPTION
**Pull request by comio**
Bug: https://bugs.gentoo.org/643398
Closes: https://bugs.gentoo.org/643398

### CODE REVIEW

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: Wrong year. We're in 2018 now.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: New ebuilds should use EAPI 6 unless there's something that currently breaks with it.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+PYTHON_COMPAT=( pypy3 python3_{4,5,6} )
+inherit eutils python-single-r1 systemd
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: Moving to EAPI6 lets us drop eutils here.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+	if use etc-crontab-systemd
+	then	sed -i \
+			-e "s!/etc/crontab!/etc/crontab-systemd!" \
+			-- "${S}/src/man/crontab."{1,5}".in" \
+			"${S}/src/bin/systemd-crontab-generator.py" || die
+	fi
+
+	epatch "${FILESDIR}"/${P}-fix-generator-path.patch
+	epatch_user
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: If EAPI 6 is used, this needs to be eapply{,_user} instead.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		--mandir="${EPREFIX}/usr/share/man" \
+		--unitdir="$(systemd_get_unitdir)" \
+		--generatordir="$(systemd_get_systemgeneratordir)" \
+		$(my_use_enable cron-boot boot) \
+		$(my_use_enable minutely) \
+		$(my_use_enable yearly) \
+		$(my_use_enable yearly quarterly) \
+		$(my_use_enable yearly semi_annually) \
+		$(my_use_enable setgid) \
+		--enable-persistent=yes
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: Optional: Move this up with the other fixed options.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+IUSE="cron-boot etc-crontab-systemd minutely setgid test yearly"
+
+RDEPEND=">=sys-apps/systemd-217
+	     sys-apps/debianutils
+	     !etc-crontab-systemd? ( !sys-process/dcron )
+	     ${PYTHON_DEPS}
+		 sys-process/cronbase"
+
+DEPEND="sys-process/cronbase
+	test? ( sys-apps/man-db dev-python/pyflakes )"
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: One dependency per line, please. It makes it a bit easier to read the list.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="cron-boot etc-crontab-systemd minutely setgid test yearly"
+
+RDEPEND=">=sys-apps/systemd-217
+	     sys-apps/debianutils
+	     !etc-crontab-systemd? ( !sys-process/dcron )
+	     ${PYTHON_DEPS}
+		 sys-process/cronbase"
+
+DEPEND="sys-process/cronbase
```
Comment (titanofold) [inline sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: Throw sys-process/cronbase into a CDEPEND or COMMON_DEPEND variable, and then use that variable in DEPEND and RDEPEND.

File: sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild (hunk 5 above)
Comment (comio) [reply sys-process/systemd-cron/systemd-cron-1.5.12-r1.ebuild]: this is out of scope of this patch.
Anyway, I will try to produce an EAPI=6 ebuild.

**Comment by floppym** [comment]
Oops, I guess I missed that. Could you please submit this change upstream?

**Comment by floppym** [comment]
Please submit the change upstream and give a link to the PR on the bug report.

**Pull request closed without merging**